sql-plugin/src/main/scala/org/apache/spark/sql/rapids/predicates.scala

```diff
@@ -526,12 +527,12 @@ case class GpuLessThan(left: Expression, right: Expression) extends CudfBinaryCo
     }
   }
 
-  override def doColumnar(lhs: GpuColumnVector, rhs: Scalar): ColumnVector = {
+  override def doColumnar(lhs: GpuColumnVector, rhs: GpuScalar): ColumnVector = {
     val result = super.doColumnar(lhs, rhs)
     if (hasFloatingPointInputs) {
       withResource(result) { result =>
         withResource(lhs.getBase.isNotNan) { lhsNotNan =>
-          withResource(GpuScalar.from(GpuScalar.isNan(rhs))) { rhsNan =>
+          withResource(GpuScalar.from(rhs.isNan, BooleanType)) { rhsNan =>
```

Comment:
```suggestion
          withResource(Scalar.fromBool(rhs.isNan)) { rhsNan =>
```

This comment applies to lots of places in this file.

Reply:
Done